scripts/src/main/resources/scripts/command/npm

```diff
@@ -16,13 +16,47 @@ source "$(dirname "${0}")"/../functions
 
 # $1: optional setup
 function doSetup() {
-  if [ ! -x "${NPM_CMD}" ]
+  local software_version=${NPM_VERSION}
+  local installed_version
+  if [ -x "${NPM_CMD}" ]
   then
+    installed_version=$(${NPM_CMD} -version)
+  else 
     doDevonCommand node setup silent
-  elif [ "${1}" != "silent" ]
-  then
-    doEcho "npm is already installed at ${NPM_CMD}"
+    installed_version=$(${NPM_CMD} -version)
   fi
+
+  if [ -z "$software_version" ] || [ "$installed_version" == "$software_version" ] 
+  then
+    if [ "${1}" != "silent" ]
+    then
+      doEcho "Version ${installed_version} of npm is already installed at ${NPM_CMD}"
+    fi
+  else
+    local download_url="https://registry.npmjs.org/npm/-/npm-${software_version}.tgz"
+    local software_dir="${NODE_HOME}/node_modules/npm"
+    doInstall "${software_dir}" "${download_url}" "npm" "${software_version}"
+    if [ -f "${NODE_HOME}/npm" ]
+    then 
+      rm "${NODE_HOME}/npm"
+    fi    
+    if [ -f "${NODE_HOME}/npm.cmd" ]
+    then 
+      rm "${NODE_HOME}/npm.cmd"
+    fi
+    if [ -f "${NODE_HOME}/npx" ]
+    then 
+      rm "${NODE_HOME}/npx"
+    fi    
+    if [ -f "${NODE_HOME}/npx.cmd" ]
+    then 
+      rm "${NODE_HOME}/npx.cmd"
+    fi
+    cp "${software_dir}/bin/npm" "${NODE_HOME}"
+    cp "${software_dir}/bin/npm.cmd" "${NODE_HOME}"
+    cp "${software_dir}/bin/npx" "${NODE_HOME}"
+    cp "${software_dir}/bin/npx.cmd" "${NODE_HOME}"
```

Comment:
Instead of removing and copying, I would prefer to create symlinks but due to windows limitations this does not seem to be an option. So I think there is no better solution than yours here that should always work (KISS).

Reply:
For the record: the current approach could fail on Windows if a `npm.cmd` is currently running in parallel due to file locking.